tests/python/relay/test_any.py

```diff
@@ -18,27 +18,156 @@
 
 import tvm
 from tvm import relay
-from tvm.relay import Kind, transform
 from tvm.relay.loops import while_loop
 from tvm.relay.testing import run_infer_type as infer_type
 
 def int32(val):
     return relay.const(val, 'int32')
 
+def any_dims(ndim):
+    shape = []
+    for _ in range(ndim):
+        shape.append(relay.Any())
```

Comment:
It would be more convenient to have dimensions with -1 internally map to relay.Any rather than requiring it to be explicit. This would match the syntax of all other frameworks when it comes to dynamic shape.

Reply:
I see the argument for having convenient ways to specify dynamic shapes from the frontends, but for analysis and transformation having dynamic shapes be treated specially is important instead of overloading integers. Annoyingly some of the Relay operators have the semantics of MxNet's operators baked in right now. My 2c is that we should make greater use of dialects of Relay in which the operators macro expand to the ones that are easy to operate and analyze.

Reply:
IMO we should remove those ad-hoc semantics like those in reshape.